src/Symfony/Component/ErrorHandler/Tests/Command/ErrorDumpCommandTest.php

```diff
@@ -0,0 +1,142 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ErrorHandler\Tests\Command;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use Symfony\Bundle\FrameworkBundle\Console\Application;
+use Symfony\Bundle\TwigBundle\Tests\TestCase;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\ErrorHandler\Command\ErrorDumpCommand;
+use Symfony\Component\ErrorHandler\ErrorRenderer\ErrorRendererInterface;
+use Symfony\Component\ErrorHandler\Exception\FlattenException;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Symfony\WebpackEncoreBundle\Asset\EntrypointLookupInterface;
+
+class ErrorDumpCommandTest extends TestCase
+{
+    private string $tmpDir = '';
+
+    public function testDumpPages()
+    {
+        mkdir($this->tmpDir);
+        touch($this->tmpDir.\DIRECTORY_SEPARATOR.'400.html');
+
+        $tester = $this->getCommandTester($this->getKernel(), []);
+        $tester->execute([]);
+
+        $this->assertFileExists($this->tmpDir.\DIRECTORY_SEPARATOR.'400.html');
+        $this->assertFileExists($this->tmpDir.\DIRECTORY_SEPARATOR.'404.html');
+        $this->assertStringContainsString('Error 404', file_get_contents($this->tmpDir.\DIRECTORY_SEPARATOR.'404.html'));
+    }
+
+    public function testDumpPagesOnlyForGivenStatusCodes()
+    {
+        $tester = $this->getCommandTester($this->getKernel());
+        $tester->execute([
+            'status-codes' => ['400', '500'],
+        ]);
+
+        $this->assertFileDoesNotExist($this->tmpDir.\DIRECTORY_SEPARATOR.'404.html');
+
+        $this->assertFileExists($this->tmpDir.\DIRECTORY_SEPARATOR.'400.html');
+        $this->assertStringContainsString('Error 400', file_get_contents($this->tmpDir.\DIRECTORY_SEPARATOR.'400.html'));
+    }
+
+    public function testDumpPagesAtCustomPath()
+    {
+        $path = sys_get_temp_dir().'/error_pages_custom';
+
+        $fs = new Filesystem();
+        $fs->remove($path);
+
+        $tester = $this->getCommandTester($this->getKernel());
+        $tester->execute([
+            '--path' => $path,
+        ]);
+
+        $this->assertFileExists($path.\DIRECTORY_SEPARATOR.'400.html');
+        $this->assertStringContainsString('Error 400', file_get_contents($path.\DIRECTORY_SEPARATOR.'400.html'));
+    }
+
+    public function testClearPages()
+    {
+        mkdir($this->tmpDir);
+        touch($this->tmpDir.\DIRECTORY_SEPARATOR.'test.html');
+
+        $tester = $this->getCommandTester($this->getKernel());
+        $tester->execute([
+            '--clear' => true,
+        ]);
+
+        $this->assertFileDoesNotExist($this->tmpDir.\DIRECTORY_SEPARATOR.'test.html');
+        $this->assertFileExists($this->tmpDir.\DIRECTORY_SEPARATOR.'404.html');
+    }
+
+    protected function setUp(): void
```

Comment:
We usually put the setup method at the beginning of the class

Reply:
Even if it is protected? :thinking: I usually put public methods first (or let PHP-CS-Fixer fix that for me)

Reply:
yes